### DISCUSSION TITLE
Switch stream

### DISCUSSION THREAD

**Discussion by flex23sky** [ant-media/Ant-Media-Server#6679] [Q&A]
I have 3 ip cameras and i would like fast switch between stream

How to implement switch stream without (minimal) latency in webapp?

Current my implementation is not fast

```
useEffect(() => {
    webrtcStreamStore.play(data.streamId) // >> adaptor.play

    return () => {
      webrtcStreamStore.stop(data.streamId) // >> adaptor.stop
    }
  }, [roomId])
```

**Comment by SelimEmre** [comment]
Hi @flex23sky,

If you only have 3 IP cameras and data transfer cost is not a major concern, I would recommend playing all 3 simultaneously. You can show or hide them as needed, and switch between them using the HTML video tag to control their visibility.

**Comment by burak-58** [comment]
Hi @flex23sky for the question and @SelimEmre for the contribution.

@flex23sky can you give more info for the scenario? Are you playing with WebRTC or HLS?
If it is WebRTC you may check multitrack playback.

Also let me convert this to a github discussions.

**Comment by flex23sky** [comment]
Hi, @SelimEmre and @burak-58 

Use WebRTC streams

Ant media version: "versionName": "2.10.0"

I published 2 WebRTC streams: 'main' and 'sub1', and I wanted to make 'sub1' a subtrack using the method
`curl -X 'POST' 'https://<my_media.com>:5443/WebRTCAppEE/rest/v2/broadcasts/main/subtrack?id=sub1' -H 'accept: application/json',`
but I get the following error:

The server encountered an unexpected condition that prevented it from fulfilling the request
Exception: 
`jakarta.servlet.ServletException: java.lang.NullPointerException: Cannot invoke &quot;String.equals(Object)&quot; because the return value of &quot;io.antmedia.datastore.db.types.Broadcast.getConferenceMode()&quot; is null`

If I connect to 2 streams simultaneously on the client side and display the required stream depending on the room while pausing the other stream on the client, there will be high network load, and this will increase as the system scales
